rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+- Route
+- Vehicle
+- Operator(s)
+- Headsign
+- Block
+- Run
+- Various service adjustments or interventions
+- Scheduled stop times
+- Predicted stop times
+- Actual stop times (in the past for historical instances of trips)
+
+This list is not exhaustive, and in some applications these other entities or data points may be accessed by joining via other parts of the data model as opposed to looking directly at a trip, but hopefully it impresses the importance getting trips right.
+
+## Present shortcomings
+
+Currently, however, our model of trip data on light rail leaves a great deal to be desired. For historical reasons, MBTA light rail vehicles have had neither on-vehicle systems by which an operator can assign their vehicle to a particular run (as in the case with bus) or a centralized dispatch system with its own model of trips and trip assignment (as with heavy rail). Furthermore, inconsistencies in how light rail trips are modeled across systems, related to peculiarities like having both a lead car and a trailer car that both need to have an operator scheduled, lead to more difficulty in exchanging data across systems. This situation creates a variety of tangible problems:
+
+1. Glides needs to perform complicated logic when importing schedule data from HASTUS to transform it into a model that matches the Glides trainsheet interface.
+1. Our passenger information (RTR) and dispatch (Glides) applications both implement their own logic to associate vehicles with trips, duplicating effort across teams while also creating additional confusion.
+1. Glides does not publish trainsheet changes using trip IDs that directly correspond to the trip IDs used by GTFS, further complicating the logic in RTR that has to assign trips to vehicles.
+1. Because RTR has to guess as to the correct trip to assign to a train, trip IDs published in our predictions are often not very sensible, or we give up on guessing a scheduled trip completely and assign a train to an added trip. As a result, passenger-facing systems such as mbta.com or screens that may wish to display both scheduled and predicted departure information can't always be sure if two trips with different IDs in the schedule versus the realtime data are actually "the same." Showing scheduled and realtime data for the same trip side-by-side is also a common pattern in third-party applications, so when a trip fails to match to the schedule the result is a combination of a scheduled trip without realtime tracking and a realtime arrival prediction without a corresponding scheduled time, creating a confusing experience for users.
+1. Analysis of historical data for performance purposes is also complicated by the difficulty in linking a trip from the schedule with a trip from the (historical) realtime data.
```

Comment:
Analysis is also currently complicated by the difficulty of linking a `tripUpdate` with a trip using data from LAMP. This makes it harder to answer questions like "How many trip have notes, and what types of notes are they"?

Reply:
Good point - I had had LAMP in mind when writing this but didn't explicitly call it out. I added a note in 1dac211.